openedx/core/djangoapps/user_api/management/commands/email_opt_in_list.py

```diff
@@ -255,6 +256,17 @@ def _write_email_opt_in_prefs(self, file_handle, org_aliases, courses):
         row_count = 0
         for row in self._iterate_results(cursor):
             user_id, username, email, full_name, course_id, is_opted_in, pref_set_datetime = row
+
+            if pref_set_datetime:
+                # TODO: Remove Django 1.11 upgrade shim
+                # SHIM: pref_set_datetime.tzinfo should always be None here after the 1.11 upgrade
+                # As of Django 1.9 datetimes returned from raw sql queries are no longer coerced to being tz aware
+                # so we correct for that here.
+                if pref_set_datetime.tzinfo is None or pref_set_datetime.tzinfo.utcoffset(pref_set_datetime) is None:
```

Comment:
Perhaps this condition could be simplified to:

```python
if pref_set_datetime.utcoffset() is None:
```

since it implicitly returns `None` for tz unaware datetimes?

Reply:
I believe that if either the tzinfo or the utcoffset is set the timestamp will be considered tz aware (and the following line will fail), and that it's possible to have a timestamp with a tzinfo but no utcoffset and vice versa so I think we need to check both. At least that's my takeaway from [the docs](https://docs.python.org/2/library/datetime.html). It's possible that the pre-1.9 Django shims enforce something more strict when setting the timezone, but this seemed the safest way to go.